### PR TITLE
VM: Bus allocation comments, constants and var naming

### PR DESCRIPTION
**Pull request by tomponline**
Some small clarification comments, and tweaks to constants and var naming for clarity.

Also adds `qemuSerialChardevName` constant to share serial char dev name between qemu and qmp to avoid hardcoding it in two places.

### CODE REVIEW

File: lxd/instance/drivers/driver_qemu.go [hunk 1]
```diff
@@ -1588,8 +1588,11 @@ func (vm *qemu) generateQemuConfigFile(busName string, devConfs []*deviceConfig.
 	// Setup the bus allocator.
 	bus := qemuNewBus(busName, sb)
 
-	// Now add the fixed set of devices.
-	devBus, devAddr, multi := bus.allocate("generic")
+	// Now add the fixed set of devices. The multi-function groups used for these fixed internal devices are
+	// specifically chosen to ensure that we consume exactly 4 PCI bus ports (on PCIe bus). This ensures that
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu.go:1592]: @stgraber I thought adding a comment about why the specific bus function groups have been picked here would be useful in the future, although I'm not sure I've got the reasoning 100% correct given the different architectures we support.

File: lxd/instance/drivers/driver_qemu.go [hunk 2]
```diff
@@ -1704,6 +1707,10 @@ func (vm *qemu) generateQemuConfigFile(busName string, devConfs []*deviceConfig.
 
 	// Record the mounts we are going to do inside the VM using the agent.
 	agentMounts := []instancetype.VMAgentMount{}
+
+	// These devices are sorted so that NICs are added first to ensure that the first NIC can use the 5th
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu.go:1711]: @stgraber same concern here as https://github.com/lxc/lxd/pull/7525/files#r439429739

File: lxd/instance/drivers/driver_qemu.go (hunk 1 above)
Comment (stgraber) [reply lxd/instance/drivers/driver_qemu.go:1592]: It's also meant to group all low-bandwidth internal devices onto a single address.

We can plug a total of 256 things into a PCIe bus, but this assumes 32 chassis * 8 function. If we waste chassis with internal devices, we'll end up seriously limiting the number of network interfaces or vfio devices that can be attached as those need a full chassis for themselves.

File: lxd/instance/drivers/driver_qemu.go (hunk 2 above)
Comment (stgraber) [reply lxd/instance/drivers/driver_qemu.go:1711]: Yeah, that's only true on PCIe but even on non-PCIe, having NICs first means that their names won't change when you add a disk, gpu, ...

File: lxd/instance/drivers/driver_qemu.go (hunk 1 above)
Comment (tomponline) [reply lxd/instance/drivers/driver_qemu.go:1592]: @stgraber thats great, will update comment to save knowledge/design reasoning.

**Comment by tomponline** [comment]
Updated

**Comment by tomponline** [comment]
Fixed tests

**Pull request merged**